src/utils/gateway.ts

```diff
@@ -31,3 +34,45 @@ export const getExplorerLink = (
 export const isWalletConnectSafeApp = (url: string): boolean => {
   return url === WC_APP_PROD.url || url === WC_APP_DEV.url
 }
+
+const signTxServiceMessage = async (
+  chainId: string,
+  safeAddress: string,
+  safeTxHash: string,
+  signer: JsonRpcSigner,
+): Promise<string> => {
+  const domain = {
+    name: 'Safe Transaction Service',
+    version: '1.0',
+    chainId: chainId,
+    verifyingContract: safeAddress,
+  }
+  const types = {
+    DeleteRequest: [
+      { name: 'safeTxHash', type: 'bytes32' },
+      { name: 'totp', type: 'uint256' },
+    ],
+  }
+  const message = {
+    safeTxHash: safeTxHash,
+    totp: Math.round(Date.now() / 3600e3),
```

Comment:
We have to `Math.floor` here according to the algorithm (https://en.wikipedia.org/wiki/Time-based_one-time_password)